### ISSUE TITLE
Can main thread be vCPU? Or we must new std::thread as vCPU?

### ISSUE THREAD

**Issue by jiangdongzi** [alibaba/PhotonLibOS#493]

**Comment by beef9999** [comment]
yes, main thread can become vCPU, as long as you have called photon::init

However, main thread/vCPU is not included in the WorkPool, even workpool is usually created from the main thread

**Comment by Coldwings** [comment]
Actually, any `std::thread` or `pthread` that called `photon::init` or `photon::vcpu_init`  will be treated as a vCPU, so yes, if you initialized photon environment by calling `photon::init` or `photon::vcpu_init`, photon will turns it into vCPU.

WorkPool is another utility, is a wrapped tool for creating thread, turns into vCPU ( or make some vCPU joins as a worker), and deploy works.
